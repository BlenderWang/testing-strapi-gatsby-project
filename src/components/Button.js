import React from "react"

export default function Button(props) {
  return (
    <div>
      <button className="btn btn--dark" onClick={props.handleClick}>
        {props.btnTxt}
      </button>
    </div>
  )
}
