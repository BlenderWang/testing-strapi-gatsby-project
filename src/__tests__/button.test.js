import React from 'react'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json'
import { shallow, render } from 'enzyme'
import Button from '../components/Button'

configure({ adapter: new Adapter() });

describe('test btn component', () => {
    it('should render btn component properly in DOM', () => {
        shallow(<Button btnTxt={'test'} />)
    })

    it('should find btn html element', () => {
        const wrapper = shallow(<Button btnTxt={'test'} />)
        expect(wrapper.find('button')).toHaveLength(1)
    })

    it('should find btn html element with classname of btn', () => {
        const wrapper = shallow(<Button className='btn' btnTxt={'test'} />)
        expect(wrapper.find('button.btn')).toBeTruthy()
    })

    it('test click event', () => {
        const simulateCb = jest.fn()

        const wrapper = shallow((<Button onClick={simulateCb} className='btn' btnTxt={'test'} />))
        wrapper.find('button').simulate('click')
        expect(wrapper.find('button').hasClass('btn')).toEqual(true)
    })
})

test('should create a snapshot', () => {
    const wrapper = shallow((<Button onClick={() => {}} className='btn' btnTxt={'test'} />))
    expect(toJSON(wrapper)).toMatchSnapshot('click')
})

test('should create a snapshot', () => {
    const wrapper = render((<Button onClick={() => {}} className='btn' btnTxt={'test'} />))
    expect(toJSON(wrapper)).toMatchSnapshot()
})
