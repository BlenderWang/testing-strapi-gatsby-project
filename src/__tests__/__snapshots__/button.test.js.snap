// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a snapshot 1`] = `
<div>
  <button
    class="btn btn--dark"
  >
    test
  </button>
</div>
`;

exports[`should create a snapshot: click 1`] = `
<div>
  <button
    className="btn btn--dark"
  >
    test
  </button>
</div>
`;
